src/components/Board/Board.js

```diff
@@ -0,0 +1,261 @@
+import React, { Component } from 'react'
+import Cell from '../Cell/Cell';
+import Row from '../Row/Row'
+
+
+class Board extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state= {table: this.createtable(props), color : 'solid'};
+        
+        this.open = this.open.bind(this);
+        this.openCleanSpaces = this.openCleanSpaces.bind(this);
+    }
+
+    createtable = props => {
+        let board = [];
+        
+        for (let i = 0; i < props.rows; i++){
+            board.push([]);
+
+            for (let j = 0; j < props.columns; j++){
+                board[i].push({
+                    x: j,
+                    y: i,
+                    count: 0,
+                    isOpen: false,
+                    hasMine: false,
+                    flags: props.flags,
+                    hasFlag: false
+                });
+            }
+
+//add the mines to the table
+        }
+        const mines = []
+
+        for (let i = 0; i < props.mines && mines.length<= props.mines; i++){
+            let randomRow = Math.floor(Math.random()*props.rows);
+            let randomColumn = Math.floor(Math.random() * props.columns);
+            let cell = board[randomRow][randomColumn];
+            if (!cell.hasMine){
+                mines.push(cell);
+                cell.hasMine = true;
+            }
+        }
+
+//add the count (number of adjacent mines arount) to the cell propety
+        let r = props.rows;
+        let c = props.columns;
+        let dir = [[1,0], [-1,0], [0,1], [0,-1], [-1,-1], [-1,1], [1,1], [1,-1]];
+
+        function valid(i,j){
+            if (i>=0 && j>=0 && i < r && j < c){
+                return true;
+            }
+            return false;
+        }
+
+
+        function find(i,j,board){
+            let s = 0;
+            for (let k = 0; k < 8; k ++){
+                let ni = i + dir[k][0];
+                let nj = j + dir[k][1];
+                if (valid (ni,nj))
+                    if (board[ni][nj].hasMine)
+                        s+=1;
+            }
+            return s;
+        }
+
+        
+        function findsomeofneighbors(board){
+
+            for (let i = 0;  i < r; i++){
+                for (let j = 0; j< c; j++)
+                {
+                    board[i][j].count = find(i,j,board)
+                }
+            }
+        }
+
+        findsomeofneighbors(board);
+        return board;
+    };
+
+
+
+
+
+
+
+
+    // a function that gets a cell and incremaens the number of flags and update the boolean of this cel
+    handleLeftClick = Cell =>
+   {
+         Cell.isOpen = true;
+         let newFlags = this.props.flags+ +1 
+         this.setState({flags:newFlags});
+
+        
+   }
+
+        
+
+
+    //a function that opens all the cells around a cell (if has no mines and its not a flag)
+
+   openCleanSpaces = Cell => {
```

Comment:
Fix indentations (tabs) here to make your code look better

Reply:
Also - split this function into smaller ones, and document this code to make it look better.
Each logical part here should be a different function.